From 9b8d8d0d62549996a9da530d56cd14f9cce9c73d Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Thu, 25 Oct 2018 11:58:40 +0530
Subject: [PATCH 0478/1851] v4l: xilinx-multi-scaler: Noncontiguous planes
 support

commit 9891ec920654281d4c2eea696654f730eea33ac8 from
https://github.com/Xilinx/linux-xlnx.git

By default, gstreamer preferred non-contiguous planes
and the multi-scaler also supports non-contiguous planes.

The patch replaces contiguous V4L2 fourcc to
noncontiguous fourcc.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 9fc097f2867e..e500d07aae56 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -175,13 +175,13 @@ static const struct xm2msc_fmt formats[] = {
 	},
 	{
 		.name = "nv16",
-		.fourcc = V4L2_PIX_FMT_NV16,
+		.fourcc = V4L2_PIX_FMT_NV16M,
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV8,
 		.num_planes = 2,
 	},
 	{
 		.name = "nv12",
-		.fourcc = V4L2_PIX_FMT_NV12,
+		.fourcc = V4L2_PIX_FMT_NV12M,
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV8_420,
 		.num_planes = 2,
 	},
@@ -199,13 +199,13 @@ static const struct xm2msc_fmt formats[] = {
 	},
 	{
 		.name = "xv20",
-		.fourcc = V4L2_PIX_FMT_XV20,
+		.fourcc = V4L2_PIX_FMT_XV20M,
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV10,
 		.num_planes = 2,
 	},
 	{
 		.name = "xv15",
-		.fourcc = V4L2_PIX_FMT_XV15,
+		.fourcc = V4L2_PIX_FMT_XV15M,
 		.xm2msc_fmt = XILINX_M2MSC_FMT_Y_UV10_420,
 		.num_planes = 2,
 	},
-- 
2.31.1

